common.go

```diff
@@ -328,15 +328,13 @@ func getInterpreter(data []byte) (interpreter string) {
 	return
 }
 
-func getFirstLine(data []byte) []byte {
-	buf := bufio.NewScanner(bytes.NewReader(data))
-	buf.Scan()
-	line := buf.Bytes()
-	if err := buf.Err(); err != nil {
-		return nil
+func getFirstLine(content []byte) []byte {
+	nlpos := bytes.IndexByte(content, '\n')
+	if nlpos < 0 {
+		nlpos = len(content)
```

Comment:
No need for additional step if no new line:
```suggestion
		return content
```